Library/Homebrew/software_spec.rb

```diff
@@ -234,7 +234,8 @@ def initialize(formula, spec)
   end
 
   def compatible_cellar?
-    cellar == :any || cellar == HOMEBREW_CELLAR.to_s
+    cellar == :any || cellar == HOMEBREW_CELLAR.to_s ||
+    (cellar != :any && File.directory?(cellar) && File.realpath(cellar) == HOMEBREW_CELLAR.to_s)
```

Comment:
Is `cellar != :any` necessary here? Isn't that already established by the first clause?

Ruby 1.8 doesn't have `File.realpath`. Do we need `realpath`, or can we simply resolve one symlink?